src/containers/admin/sideBar/SideBar.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Menu } from 'antd';
+import { useNavigate } from 'react-router-dom'
+import PubSub from 'pubsub-js';
+import menuList from '../../../config/menuConfig'
+import logo from '../../../static/imgs/EpamLogo.jpeg'
+import './css/sidebar.less'
+
+
+//Functions of antD
+function getItem(label, key, icon, children, type) {
+    return {
+        key,
+        icon,
+        children,
+        label,
+        type,
+    };
+}
+
+//Own function: Making a list by recursion
+let getList = (menuList) => {
+    return menuList.reduce((pre, cur) => {
+        if (!cur.children) {
+            let resItem = getItem(cur.title, cur.key, cur.icon)
+            return [...pre, resItem]
+        } else {
+            let resItem = getItem(cur.title, cur.key, cur.icon,
+                getList(cur.children)
+            )
+            return [...pre, resItem]
+        }
+    }, [])
+}
+const items = getList(menuList)  //[{...},{...}]
+
+
+export default function SideBar() {
+    const navigate = useNavigate()
+    const {pathname} = window.location
+
+    //get "getTitle" from pubsub
+    let getTitle;
+    PubSub.subscribe('gTitle', (_, data) => {
```

Comment:
If you just want to change title of the Header after navigating. You can use `useLocation` in Header component to get notified whenever URL changed, no need to use PubSub here.